terraform/modules/shared/main.tf

```diff
@@ -513,6 +513,37 @@ resource "azurerm_container_app_environment_storage" "custom_schema_storage" {
   access_mode                  = "ReadWrite"
 }
 
+##### FHIR Server #####
+
+resource "azurerm_healthcare_service" "fhir_server" {
+  name                = "${terraform.workspace}fhir${substr(var.client_id, 0, 8)}"
+  location            = "eastus"
+  resource_group_name = var.resource_group_name
+  kind                = "fhir-R4"
+  cosmosdb_throughput = (terraform.workspace == "uat" ? 2000 : 400)
+
+  lifecycle {
+    ignore_changes = [name, tags]
+  }
+
+  tags = {
+    environment = terraform.workspace
+    managed-by  = "terraform"
+  }
+}
+
```

Comment:
This is how we did things previously, but currently Azure has two managed FHIR servers:

1. [Azure API for FHIR](https://learn.microsoft.com/en-us/azure/healthcare-apis/azure-api-for-fhir/get-started-with-azure-api-fhir)
2. [Azure Health Data Services FHIR service](https://learn.microsoft.com/en-us/azure/healthcare-apis/healthcare-apis-overview)

Currently we are using Azure API for FHIR, but this will be deprecated on 09-30-2026 and doesn't offer autoscaling out of the box. I think we should upgrade now to Azure Health Data Services FHIR service so that LAC does not need to handle a [migration](https://learn.microsoft.com/en-us/azure/healthcare-apis/fhir/migration-strategies) in the future and so we can take advantage of native autoscaling now for both performance and cost savings.

I believe [these](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/healthcare_fhir_service) are the relevant Terraform docs